tools/genomespace/genomespace_importer.py

```diff
@@ -1,220 +1,265 @@
-# Dan Blankenberg
-
+import argparse
 import json
-import optparse
 import os
-import shutil
-import tempfile
+import sys
+import uuid
 
-from six.moves import http_cookiejar
-from six.moves.urllib.parse import parse_qs, unquote_plus, urlparse
-from six.moves.urllib.request import build_opener, HTTPCookieProcessor, Request, urlopen
+from genomespaceclient import GenomeSpaceClient
 
+import galaxy
 from galaxy.datatypes import sniff
 from galaxy.datatypes.registry import Registry
 
-GENOMESPACE_API_VERSION_STRING = "v1.0"
-GENOMESPACE_SERVER_URL_PROPERTIES = "https://dm.genomespace.org/config/%s/serverurl.properties" % ( GENOMESPACE_API_VERSION_STRING )
-DEFAULT_GENOMESPACE_TOOLNAME = 'Galaxy'
-FILENAME_VALID_CHARS = '.-()[]0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ '
-
-CHUNK_SIZE = 2**20  # 1mb
-
-DEFAULT_GALAXY_EXT = "data"
 
-# genomespace format identifier is the URL
-GENOMESPACE_FORMAT_IDENTIFIER_TO_GENOMESPACE_EXT = {}  # TODO: fix this so it is not a global variable
-# TODO: we should use a better way to set up this mapping
+# Mappings for known genomespace formats to galaxy formats
 GENOMESPACE_EXT_TO_GALAXY_EXT = {'rifles': 'rifles',
                                  'lifes': 'lifes',
                                  'cn': 'cn',
-                                 'GTF': 'gtf',
+                                 'gtf': 'gtf',
                                  'res': 'res',
                                  'xcn': 'xcn',
                                  'lowercasetxt': 'lowercasetxt',
                                  'bed': 'bed',
-                                 'CBS': 'cbs',
+                                 'cbs': 'cbs',
                                  'genomicatab': 'genomicatab',
                                  'gxp': 'gxp',
                                  'reversedtxt': 'reversedtxt',
                                  'nowhitespace': 'nowhitespace',
                                  'unknown': 'unknown',
                                  'txt': 'txt',
                                  'uppercasetxt': 'uppercasetxt',
-                                 'GISTIC': 'gistic',
-                                 'GFF': 'gff',
+                                 'gistic': 'gistic',
+                                 'gff': 'gff',
                                  'gmt': 'gmt',
-                                 'gct': 'gct'}
+                                 'gct': 'gct',
+                                 'fasta': 'fasta',
+                                 'fna': 'fasta',
+                                 'fa': 'fasta',
+                                 'faa': 'fasta',
+                                 'ffn': 'fasta',
+                                 'fastq': 'fastqsanger',
+                                 'fq': 'fastqsanger',
+                                 'fqz': 'fastqsanger'}
```

Comment:
@mvdbeek Thanks for reviewing. I've kept the original implementation logic as is, and have only refactored it. The specifying line where filetype is determined is in the determine_file_type() method here: https://github.com/galaxyproject/galaxy/pull/1814/files/e44d67f1944aa43c9d28cd9efffca88cfecd51d5#diff-a9715b311879402b8b86a4013e4ec56bR152

The order of resolution is:
1. Respect genomespace supplied type
2. If no type, try to sniff type
3. Still no type, use filename extension
4. Still nothing, default to 'data'

There's no forced coercion unless the file is misnamed. Does that sound ok?

However, one thing I did change was to call sniff.handle_uploaded_dataset() to trigger decompression here prior to determining file type: https://github.com/galaxyproject/galaxy/pull/1814/files/e44d67f1944aa43c9d28cd9efffca88cfecd51d5#diff-a9715b311879402b8b86a4013e4ec56bR208

Given the current implementation of sniff.handle_uploaded_dataset(), only fasta.gz files will be decompressed. Everything else will be left as is.